File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        i = 0
        j = 1
        output = 0

        while (j < len(prices)):
            if (prices[j] > prices[i]):
                profit = prices[j] - prices[i]
                output = max(output, profit)
            else:
                i = j
            
            j +=1
        
        return output

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
var maxProfit = function(prices) {
    let k = 0
    let i = 1
    let maxProfit = 0

    while (i < prices.length) {
        if (prices[i] > prices[k]) {
            profit = prices[i] - prices[k]
            maxProfit = Math.max(profit, maxProfit)
        }
        else {
            k = i
        }

        i ++
    }

    return maxProfit
};